src/Sarif/Core/Result.cs

```diff
@@ -117,6 +123,39 @@ public bool TryIsSuppressed(out bool isSuppressed)
             return true;
         }
 
+        public void InlineArtifacts(Run run = null)
```

Comment:
>InlineArtifacts [](start = 20, length = 15)

Remember that a result has an optional run object on it. And so if we find that, we should use it. Otherwise, this looks good.

We have another inlining problem, which is that a result needs to have a 'flattened' result message in cases where it otherwise has only messge args and an index to the rules table (which actually holds the message string). We have helpers elsewhere that provide this flattening, there's a visitor dedicated to it. Should make sure useful helper exists on Rule. btw - all of this stuff is helpful to enable optionally in the rewrite command.